File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/core/handler/ReceiveDataHandler.kt
package com.syncrhrohealth.bandpass_ble_connection.core.handler

import android.util.Log
import com.syncrhrohealth.bandpass_ble_connection.core.handler.response.ImuDataRspHandler
import com.syncrhrohealth.bandpass_ble_connection.core.model.Packet

val TAG: String = ReceiveDataHandler::class.java.simpleName

object ReceiveDataHandler {

    fun handle(packet: Packet, handler: DeviceHandler) {
//        Log.d(TAG, "handle: ${packet.header.type}")

        val typeInt = packet.header.type.toInt() and 0xFF
        when (typeInt) {
            /** Responses */
            0x03 -> {
                ImuDataRspHandler.handle(packet.data, handler)
            }

            else -> {
                Log.e(TAG, "Unknown header type: ${packet.header.type}")
            }
        }
    }
}

File: ios/Classes/Core/Handler/ReceiveDataHandler.swift
//
//  ReceiveDataHandler.swift
//  Runner
//
//  Created by MAC on 11/3/25.
//

import Foundation

struct ReceiveDataHandler {
    static func handle(packet: Packet, handler: DeviceHandler) {
        print("ReceiveDataHandler - handle: \(packet.header.commandCode)")

        guard let commandCode = packet.header.commandCode else {
            print("ReceiveDataHandler - Unknown command code: nil")
            return
        }

        switch commandCode {
        
        /// Responses
        case .packetCmdDeviceInfoResp:
            DeviceInfoRspHandler.handle(data: packet.data, handler: handler)

        case .packetCmdImuDataRsp:
            ImuDataRspHandler.handle(data: packet.data, handler: handler)

        default:
            print("ReceiveDataHandler - Unknown command code: \(commandCode)")
        }
    }
}
